python/simulation.py

```diff
@@ -933,40 +936,40 @@ class Simulation(object):
     control various parameters of the Meep computation.
     """
     def __init__(self,
-                 cell_size,
-                 resolution,
-                 geometry=None,
-                 sources=None,
-                 eps_averaging=True,
-                 dimensions=3,
-                 boundary_layers=None,
-                 symmetries=None,
-                 force_complex_fields=False,
-                 default_material=mp.Medium(),
-                 m=0,
-                 k_point=False,
-                 kz_2d="complex",
-                 extra_materials=None,
-                 material_function=None,
-                 epsilon_func=None,
-                 epsilon_input_file='',
-                 progress_interval=4,
-                 subpixel_tol=1e-4,
-                 subpixel_maxeval=100000,
-                 loop_tile_base_db=0,
-                 loop_tile_base_eh=0,
-                 ensure_periodicity=True,
-                 num_chunks=0,
-                 Courant=0.5,
-                 accurate_fields_near_cylorigin=False,
-                 filename_prefix=None,
-                 output_volume=None,
-                 output_single_precision=False,
-                 geometry_center=mp.Vector3(),
-                 force_all_components=False,
-                 split_chunks_evenly=True,
-                 chunk_layout=None,
-                 collect_stats=False):
+                 cell_size: Vector3Type = None,
+                 resolution: Union[float, int] = None,
+                 geometry: Optional[List[GeometricObject]] = None,
+                 sources: Optional[List[Source]] = None,
+                 eps_averaging: bool = True,
+                 dimensions: int = 3,
+                 boundary_layers: Optional[List[PML]] = None,
+                 symmetries: Optional[List[Symmetry]] = None,
+                 force_complex_fields: bool = False,
+                 default_material: Medium = mp.Medium(),
+                 m: int = 0,
```

Comment:
Can be a float — doesn't need to be an integer if the cell doesn't contain the origin.

Reply:
Fixed.